### PR TITLE
[W5.7][W16-2]Yeo Yi Jia, Amabel

### PR DESCRIPTION
**Pull request by iMarbles**

### CODE REVIEW

**Review by Kratious — COMMENTED** [review]
Looks great. Maybe we could further improve the sort by allowing for sorting by different fields, or even by multiple fields. Can also allow for sort by ascending and descending options for the user.

**Comment by sharan8** [comment]
Nice work, found your use of 'PersonChainedComparator' to handle multiple comparators rather interesting. Would love to see it at work when looking to compare 'Person's based on name first, then by other fields. 

Perhaps a simple boolean flag and command line input to handle ascending/descending sorting would be very useful.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Sorting all persons : `sort`
+
+Sorts a list of all persons by name, alphabetically, in the address book. +
+Format: `sort`
```
Comment (joanneong) [inline docs/UserGuide.adoc:78]: Nice update to the user guide :)

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+
```
Comment (joanneong) [inline src/seedu/addressbook/commands/SortCommand.java:10]: Redundant line spacing here :P Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

Comment (joanneong) [reply src/seedu/addressbook/commands/SortCommand.java:10]: Same for redundant spacing below in the rest of your code

File: src/seedu/addressbook/data/person/Person.java [hunk 3]
```diff
@@ -1,91 +1,90 @@
...
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(name, phone, email, address, tags);
+    }
+
+    @Override
+    public String toString() {
+        return getAsTextShowAll();
+    }
+}
```
Comment (joanneong) [inline src/seedu/addressbook/data/person/Person.java:90]: Why did you remove, then copy and pasted the same piece of code again?

File: src/seedu/addressbook/data/utils/PersonChainedComparator.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data.utils;
+
+import seedu.addressbook.data.person.Person;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public class PersonChainedComparator implements Comparator<Person> {
```
Comment (joanneong) [inline src/seedu/addressbook/data/utils/PersonChainedComparator.java:9]: It would be nice to have JavaDoc for this class - it is not immediately obvious what this class is used for :P

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (joanneong) [inline src/seedu/addressbook/parser/Parser.java:14]: Is this import statement correct? Intellij tends to automatically make the import statements into an 'import all', so do be careful about this!

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 6]
```diff
@@ -1,138 +1,177 @@
...
+    public void getAllPersons() throws Exception {
+        UniquePersonList allPersons = defaultAddressBook.getAllPersons();
+        UniquePersonList personsToCheck = new UniquePersonList(aliceBetsy, bobChaplin);
+
+        assertTrue(isIdentical(allPersons, personsToCheck));
+    }
+
+    @Test
+    public void getAllSortedPersons() throws Exception {
+        AddressBook book = new AddressBook();
```
Comment (joanneong) [inline test/java/seedu/addressbook/data/AddressBookTest.java:142]: Do you need to create new instances of the address book? What happened in the set up for the test?

**Review by joanneong — APPROVED** [review]
Good effort in general :)

However, it seems like some files are accidentally deleted then recovered/ copied and pasted, so they show up as code changes even if you did not change that part. Try to make sure that this doesn't happen in future so that unintentional changes are not included in your PRs!

Please close this PR after reading/ replying to my comments :)
